pkg/csi/service/vanilla/controller.go

```diff
@@ -2639,12 +2640,20 @@ func (c *controller) processQueryResultsListVolumes(ctx context.Context, startin
 		} else {
 			volumeType = prometheus.PrometheusBlockVolumeType
 			blockVolID := cnsVolumes[i].VolumeId.Id
+			log.Infof("adkulkarni -> cnsVolumes[i]: %+v", cnsVolumes[i])
 			nodeVMUUID, found := volumeIDToNodeUUIDMap[blockVolID]
 			if found {
 				volCounter += 1
-				//Populate csi.Volume info for the given volume
+				volumeId := blockVolID
+				migratedVolumePath, err := volumeMigrationService.GetVolumePath(ctx, blockVolID, false)
+				if err != nil {
+					log.Errorf("failed to get volumePath for migrated volume. err: %v", err)
```

Comment:
If `GetVolumePath` fails for a migrated volume, we just log an error and proceed with line: 2655. Don't we have to return error

Reply:
With the modified GetVolumePath method (passing a flag to only check the in-memory cache to see if it's an in-tree volume), we are simply going through a map and returning true or false. No error is returned. 

So, I'm not expecting the control to reach this line unless something weird like `volumeMigrationService` is not initialized. Are you referring to such an error?

Reply:
Based on Divyen's above comment, I am changing the method to return `ErrNotFound` when we don't get a cache hit. But I still don't expect an error other than that one. Can you confirm if we should return an error in ListVolumes response if we get an error other than `ErrNotFound`?